mesonbuild/interpreter/interpreter.py

```diff
@@ -1662,21 +1662,28 @@ def func_build_target(self, node, args, kwargs):
         else:
             raise InterpreterException('Unknown target_type.')
 
-    @permittedKwargs({'input', 'output', 'fallback', 'command', 'replace_string'})
     @noPosargs
-    def func_vcs_tag(self, node, args, kwargs):
-        if 'input' not in kwargs or 'output' not in kwargs:
-            raise InterpreterException('Keyword arguments input and output must exist')
-        if 'fallback' not in kwargs:
+    @typed_kwargs(
+        'vcs_tag',
+        CT_INPUT_KW,
+        CT_OUTPUT_KW,
```

Comment:
We no longer manually code that this is required, because `KwargInfo(..., required=True)` yay.

But, while CT_OUTPUT_KW is required, custom_target has optional input, so CT_INPUT_KW needs to evolve to required.

Reply:
It's still not 100% correct (though as correct as it was previously), since we really should be setting `allow_empty=False`, but we don't have a way to `evolve` that in. Probably we should change the semantics of `required=True` to mean `allow_empty=False`, but that's a separate change, and this does give us parity with the original implementation

Reply:
Ah, indeed, that does seem a bit unfortunate...

Really, once again that would be a case where we should not allow empty ever, but instead have your future thoughts for a `default()` object.

Oh well,. more typed_kwargs means we get closer to the day when that can be sanely implemented, right?

Is it worth putting in a comment here?